libraries/AC_Avoidance/AP_OADijkstra.cpp

```diff
@@ -133,27 +148,39 @@ AP_OADijkstra::AP_OADijkstra_State AP_OADijkstra::update(const Location &current
         }
     }
 
-    // rebuild path if destination has changed
-    if (!destination.same_latlon_as(_destination_prev)) {
+    // rebuild path if destination or next_destination has changed
+    if (!destination.same_latlon_as(_destination_prev) || !next_destination.same_latlon_as(_next_destination_prev)) {
         _destination_prev = destination;
+        _next_destination_prev = next_destination;
         _shortest_path_ok = false;
     }
 
     // calculate shortest path from current_loc to destination
     if (!_shortest_path_ok) {
         _shortest_path_ok = calc_shortest_path(current_loc, destination, error_id);
         if (!_shortest_path_ok) {
+            dest_to_next_dest_clear = _dest_to_next_dest_clear = false;
             report_error(error_id);
             Write_OADijkstra(DIJKSTRA_STATE_ERROR, (uint8_t)error_id, 0, 0, destination, destination);
             return DIJKSTRA_STATE_ERROR;
         }
         // start from 2nd point on path (first is the original origin)
         _path_idx_returned = 1;
+
+        // check if path from destination to next_destination intersects with a fence
+        _dest_to_next_dest_clear = false;
+        if (!next_destination.is_zero()) {
+            Vector2f seg_start, seg_end;
```

Comment:
call these seg_start_cm and seg_end_cm

Reply:
I might just leave these as-is because there are about 15 other places where we use these same variable names within Dijkstras.